### PR TITLE
releng - docker update poetry version and update ubuntu base image

### PR DESCRIPTION
**Pull request by kapilt**
addresses a policy stream docker image build issue.

https://github.com/cloud-custodian/cloud-custodian/runs/7696370693?check_suite_focus=true

### CODE REVIEW

File: docker/mailer [hunk 1]
```diff
@@ -29,7 +29,7 @@ ADD tools/c7n_openstack /src/tools/c7n_openstack
 RUN rm -R tools/c7n_openstack/tests
 
 # Install requested providers
-ARG providers="azure gcp kube openstack"
+ARG providers="gcp kube openstack azure"
```
Comment (ajkerrigan) [inline docker/mailer:32]: Assuming this is because of the dep dance around the sequential poetry installs?

File: docker/policystream [hunk 2]
```diff
@@ -30,45 +29,27 @@ ADD tools/c7n_openstack /src/tools/c7n_openstack
...
 # Install requested providers
-ARG providers="azure gcp kube openstack"
+ARG providers="gcp kube openstack azure"
 RUN . /usr/local/bin/activate && \
     for pkg in $providers; do cd tools/c7n_$pkg && \
     $HOME/.poetry/bin/poetry install && cd ../../; done
 
 RUN mkdir /output
 
-# Compile libgit2
```
Comment (ajkerrigan) [inline docker/policystream:40]: Really nice to be able to drop this :+1:

**Review by ajkerrigan — APPROVED** [review]
Looks good, thanks! :tada:

File: docker/mailer (hunk 1 above)
Comment (kapilt) [reply docker/mailer:32]: it was manually added to one of the images, to address a previous issue on dependencies for that particular image, I went ahead and just added it to the template generator when regenerating all the templates. it is unclear if its still needed re this explicit ordering.

**Pull request merged**